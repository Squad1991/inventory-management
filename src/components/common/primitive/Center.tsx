import { css } from '@compiled/react';
import React, { PropsWithChildren } from 'react';
import Box from '~src/components/common/primitive/Box';

const CenterStyles = css({
  position: 'absolute',
  top: 0,
  bottom: 0,
  left: 0,
  right: 0,
  justifyContent: 'center',
  alignItems: 'center',
  height: '100%',
});

const Center: React.FC<PropsWithChildren> = ({ children }) => {
  return <Box css={CenterStyles}>{children}</Box>;
};

export default Center;
