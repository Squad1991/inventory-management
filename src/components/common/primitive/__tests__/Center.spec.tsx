import { render } from '@testing-library/react';
import React from 'react';
import Center from '../Center';

describe('Center component', () => {
  it('should render correctly', () => {
    const { container } = render(<Center>Hello, World!</Center>);
    expect(container).toMatchSnapshot();
  });
});
