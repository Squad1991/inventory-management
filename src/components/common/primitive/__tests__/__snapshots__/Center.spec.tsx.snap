// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Center component > should render correctly 1`] = `
<div>
  <style
    data-cmpld="true"
  >
    ._1bah1h6o{justify-content:center}._4cvr1h6o{align-items:center}._4t3i1osq{height:100%}
  </style>
  <style
    data-cmpld="true"
  >
    ._1e0c1txw{display:flex}
  </style>
  <div
    class="_1e0c1txw _1bah1h6o _4cvr1h6o _4t3i1osq"
  >
    Hello, World!
  </div>
</div>
`;
